core/src/main/java/org/bitcoinj/wallet/DeterministicKeyChain.java

```diff
@@ -882,7 +915,15 @@ public boolean isFollowing() {
                 if (chain == null) {
                     // If this is not a following chain and previous was, this must be married
                     boolean isMarried = !isFollowingKey && !chains.isEmpty() && chains.get(chains.size() - 1).isFollowing();
-                    if (seed == null) {
+                    // If this has a private key but no seed, then all we know is the spending key H
+                    if (seed == null & key.hasSecretBytes())
```

Comment:
coding style ... I'm pretty sure Schildbach will ask for style consistent with the project, so `if (...) `~\n~`{`.